### ISSUE TITLE
Unable to connect Dot Pad via Bluetooth

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#18584]
### Steps to reproduce:

Reported by a user (and also mentioned by @SaschaCowley )

- Try to connect Dot Pad via Bluetooth with NVDA 2025.1.2
 
### Actual behavior:

NVDA does not see device when connected via Bluetooth

### Expected behavior:

NVDA should connect via Bluetooth (advised by user that this worked previously with an add-on provided by dot inc)

#### NVDA version:
2025.1.2

#### Windows version:

Unrecorded

**Comment by SaschaCowley** [comment]
Just to note, I haven't been able to use the DotPad with NVDA via Bluetooth not because of any problem with NVDA, but because it is disconnecting (or being disconnected) from my computer about 3 seconds after I pair it.

**Comment by socheat7396** [comment]
In the select Braille Display dialog, the DotPad should be listed among the list of Braille displays. But I saw none of it. I remember seeing it listed on NVDA 2025.1.
### Steps to reproduce:
* Turn on the DotPad.
* Bring up the select Braille display dialog.
### Actual behavior:
The DotPad device is not listed in the list of Braille displays.
### Expected behavior:
The DotPad device should be listed under select  Braille display dialog.
### NVDA logs, crash dumps and other attachments:
N/A
### System configuration
#### NVDA installed/portable/running from source:
Installed
#### Windows version:
Windows 11 24H2 (AMD64) build 26100.4770
#### Does the issue still occur after restarting your computer?
Yes

**Comment by bramd** [comment]
@SaschaCowley You should not be able to pair the Dot Pad at all using Windows' Bluetooth settings. Unfortunately the hardware uses a BLE (Bluetooth Low Energy) chip without Bluetooth Classic support. For pure BLE devices, the pairing/connecting has to be handled in the application itself and Windows has a separate BLE API for that. Also, for best results don't connect a USB cable to the data port (left port) of the Dot Pad while using it over Bluetooth.

I'm currently porting over the Bluetooth code from the add-on, could you please assign this issue to me?

**Comment by bramd** [comment]
> In the select Braille Display dialog, the DotPad should be listed among the list of Braille displays. But I saw none of it. I remember seeing it listed on NVDA 2025.1.

The Dot Pad recently gained auto detection, so if you don't connect it over USB it might not show up in NVDA alpha versions. The add-on provided by Dot still adds functionality that is not in core, including Bluetooth support and custom functionality for the tactile display of the Dot Pad.

**Comment by jage9** [comment]
I see a USB fix was mentioned for 2025.3, but I assume this is a separate issue? Wanted to check for any updates as the Dot Pad X is about to be released in the U.S. and elsewhere.

**Comment by bramd** [comment]
@jage9 A quick update on this:

- BLE integration is still being worked on
- Dot has confirmed with me that the current drivers also work for Dot Pad X
- The USB fix you mention is regarding auto detection of Dot Pad over USB which works now in NVDA core
- The advise for now is to install the add-on as provided by Dot to get the most functionality for the Dot Pad. Installing the add-on will replace the driver in core with the one from the add-on and this is supported on NVDA 2025.x

**Comment by jage9** [comment]
Thanks for the updates. Is there an updated driver that supports the 2025
NVDA versions or are you just overriding the incompatible add-o message? I
think there may be more than one add-on floating around, so maybe I have an
older one.

**Comment by bramd** [comment]
> Thanks for the updates. Is there an updated driver that supports the 2025
> NVDA versions or are you just overriding the incompatible add-o message? I
> think there may be more than one add-on floating around, so maybe I have an
> older one.

The official Dot documentation should contain a link to the latest version (0.2). I'm actively working with Dot to get this in the add-on store as well. If you have actual Dot hardware, I would really appreciate if you could test the build of PR <https://github.com/nvaccess/nvda/pull/19122> as well.
